presto-main/src/main/java/com/facebook/presto/resourcemanager/NodeTaskState.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.resourcemanager;
+
+import com.facebook.drift.annotations.ThriftConstructor;
+import com.facebook.drift.annotations.ThriftField;
+import com.facebook.drift.annotations.ThriftStruct;
+import com.facebook.presto.execution.PartitionedSplitsInfo;
+
+import java.util.Objects;
+
+import static java.util.Objects.requireNonNull;
+
+@ThriftStruct
+public class NodeTaskState
```

Comment:
Why not just use NodeStatus as it is? This looks like strict subset of NodeStatus. I can imagine in the future NodeStatus can be used by the scheduler for more complex information like number of cores, expected TTL in the case of elastic capacity, etc